### DISCUSSION TITLE
Supplemental subfigures in docx

### DISCUSSION THREAD

**Discussion by d-morrison** [quarto-dev/quarto-cli#11444] [Q&A] [answered]
### Description

I'm trying to create a set of subfigures in an appendix in docx format, and I can't quite get it to work. The following reprex works as I hoped in html format but not in docx:

````qmd
---
title: "supplemental subfigures - full"
format: 
  docx:
    echo: false
  html:
    echo: false
crossref:
  custom:
    - kind: float
      key: suppfig
      reference-prefix: Figure S
      space-before-numbering: false

---

See @suppfig-subfigs-1 and @suppfig-subfigs-2:

::: {#suppfig-subfigs layout-ncol="2"}

:::: {#suppfig-subfigs-1}

```{r}
plot(cars)
```

Cars

::::

:::: {#suppfig-subfigs-2}

```{r}
plot(pressure)
```

Pressure

::::

Charts

:::
````

HTML output (looks good):

![image](https://github.com/user-attachments/assets/8e39cb95-0854-4774-9935-475a48e54d41)

docx output (I manually added the table borders to help clarify what's happening):

![image](https://github.com/user-attachments/assets/dc5b86e3-e556-4a3e-a8b9-c1e788b2d761)

It looks like the docx file has created a second cell in the first row for the second graph, but that graph instead ends up on a second row.

This seems to be an instance of #9623 and might be solvable by #7665, but since I'm particularly interested in combining custom floats *and* subfigures in docx, I thought it might be worth creating a separate discussion for this case.

**Comment by cscheid** [comment, marked as answer]
I think your assessment wrt #9623 is correct, but #7665 doesn't come up here: knitr only gets involved in code cells, and not div crossreferences.

**Reply by d-morrison** [reply]
I mentioned #7665 because using a single code chunk for both figures (with the `#| fig-subcap:` chunk option) gets me closer to what I expected:


````qmd
---
title: "supplemental subfigures - compact"
format: 
  docx:
    echo: false
  html:
    echo: false
crossref:
  custom:
    - kind: float
      key: suppfig
      reference-prefix: Figure S
      space-before-numbering: false

---

See @suppfig-subfigs-1 and @suppfig-subfigs-2:

::: {#suppfig-subfigs}

```{r}
#| layout-ncol: 2
#| fig-subcap:
#|   - "Cars"
#|   - "Pressure"

plot(cars)
plot(pressure)
```

Charts

:::
````

![image](https://github.com/user-attachments/assets/99128090-2115-443b-9e23-7d1c48b5d42c)

This way, the layout is as expected but the crossrefs to the subfigures don't work. I was hoping that #7665 might make this approach feasible?
